practical_09/taxi_simulator.py

```diff
@@ -0,0 +1,96 @@
+"""
+Taxi simulator to allow customer to drive a taxi of their choosing
+"""
+
+from practical_09.silver_service_taxi import SilverServiceTaxi
+from practical_09.taxi import Taxi
+
+MENU = """
+q)uit, c)hoose, d)rive
+>>> """
+
+taxis = [Taxi("Prius", 100), SilverServiceTaxi("Limo", 100, 2), SilverServiceTaxi("Hummer", 200, 4)]
+
+
+def main():
+    """Present user with a list of available taxis.
+    Allow them to drive that taxi a desired distance.
+    Display cumulative bill. Repeat until user quits."""
+    total_bill = 0
+    user_choice = input(MENU)
+    while user_choice.lower() != "q":
+        if user_choice.lower() == "c":
+            taxi_choice = choose_taxi()
+        elif user_choice.lower() == "d":
+            try:
+                total_bill = drive_taxi(taxi_choice, total_bill)
+            except UnboundLocalError:
```

Comment:
This is an interesting solution. It may accidentally catch other unbound local errors though.
You could consider using a variable to keep track of the current taxi, starting out as None. Then check if the current taxi is not None (you don't need to use an exception then).